doc/source/dev_guide/plugins.rst

```diff
@@ -1,34 +1,207 @@
+additional configuration files for:
 
-Here is an
-`example <https://github.com/mraspaud/satpy-composites-plugin-example>`_ of a
-composites plugin.
+* readers
+* composites and modifiers
+* enhancements
+* writers
 
-The key is to use the same configuration directory structure as satpy and add
-a `satpy.composites` entry point in the setup.py file of the plugin:
+For basic testing and temporary configuration changes, you can follow
+the instructions in :ref:`component_configuration`. This will tell Satpy
+where to look for your custom YAML configuration files and import any Python
+code you'd like it to use for these components. However, this requires telling
+Satpy of these paths on every execution (either as an environment variable or
+by using ``satpy.config``).
 
-.. code: python
+Satpy also supports being told this information via setuptools "entry points".
+Once your custom Python package with entry points is installed Satpy will
+automatically discover it when searching for composites without the user
+needing to explicitly import your package. This has the added
+benefit of organizing your YAML configuration files and any custom python code
+into a single python package. How to structure a package in this way is
+described below.
+
+An example project showing the usage of these entry points is available at
+`this github repository <https://github.com/pytroll/satpy-composites-plugin-example>`_
+where a custom compositor is created. This repository also includes common
+configuration files and tools for writing clean code and automatically testing
+your python code.
+
+Plugin package structure
+========================
+
+The below sections will use the example package name ``satpy-myplugin``. This
+is only an example and naming a plugin package with a ``satpy-`` prefix is not
+required.
+
+A plugin package should consist of three main parts:
+
+1. ``pyproject.toml`` or ``setup.py``: These files define the metadata and
+   entry points for your package. Only one of them is needed. With only a few
+   exceptions it is recommended to use a ``pyproject.toml`` as this is the new
+   and future way Python package configuration will be supported by the ``pip``
+   package manager. See below for examples of the contents of this file.
+2. ``mypkg/etc/``: A directory of Satpy-compatible component YAML files. These
+   YAML files should be in ``readers/``, ``composites/``, ``enhancements/``,
+   and ``writers/`` directories. These YAML files must follow the Satpy naming
+   conventions for each component. For example, composites and enhancements
+   allow for sensor-specific configuration files. Satpy will collect all
+   available YAML files from all installed plugins and merge them with those
+   builtin to Satpy. The Satpy builtins will be used as a "base" configuration
+   with all external YAML files applied after.
+3. ``mypkg/``: The python package with any custom python code. This code should
```

Comment:
could it be a `mypkg.py` file?

Reply:
Technically yes, but also no. The YAML files are the only way Satpy currently discovers new components and the `etc` directory currently has to be a sub-directory of the package.